packages/general-adapter/adapter/generateThemeCss.ts

```diff
@@ -1,69 +1,77 @@
-import type {
-  Themes,
-  ExportingProps,
-  ExportingResult,
-  ValueToken,
-} from "../types";
+import type { ExportingResult } from "../types";
+import { Tokens } from "../shared/tokens";
 import { autoGeneratedFile } from "@recursica/common";
 
 export function generateThemeCss(
-  themes: Themes,
-  { outputPath }: ExportingProps,
+  { themes }: Tokens,
+  outputPath: string,
 ): ExportingResult[] {
   const themeFiles: ExportingResult[] = [];
 
-  for (const [themeName, themeTokens] of Object.entries(themes)) {
+  for (const [themeName, themeModes] of Object.entries(themes)) {
     // Skip empty themes or non-object themes
     if (
-      !themeTokens ||
-      typeof themeTokens !== "object" ||
-      Array.isArray(themeTokens)
+      !themeModes ||
+      typeof themeModes !== "object" ||
+      Array.isArray(themeModes)
     ) {
       continue;
     }
 
-    const themeCssFilename = `${themeName.toLowerCase()}-theme.css`;
-    const themeCssPath = outputPath + "/" + themeCssFilename;
+    // Process each mode (Light, Dark, etc.) within the theme
+    for (const [modeName, themeTokens] of Object.entries(themeModes)) {
+      // Skip empty mode tokens or non-object mode tokens
+      if (
+        !themeTokens ||
+        typeof themeTokens !== "object" ||
+        Array.isArray(themeTokens)
+      ) {
+        continue;
+      }
 
-    // Convert theme token names to CSS custom property format
-    const cssVariables = Object.entries(themeTokens)
-      .map(([key, value]) => {
-        const cssVarName = key.replace(/\//g, "-").toLowerCase();
+      const themeCssFilename = `${themeName.toLowerCase()}-${modeName.toLowerCase()}-theme.css`;
+      const themeCssPath = outputPath + "/" + themeCssFilename;
 
-        // Handle different value types
-        if (typeof value === "string" || typeof value === "number") {
-          return `  --${cssVarName}: ${value};`;
-        }
+      // Convert theme token names to CSS custom property format
+      const cssVariables = Object.entries(themeTokens)
+        .map(([key, value]) => {
+          const cssVarName = key.replace(/[\/\s]/g, "-").toLowerCase();
 
-        // Handle object references (like { collection: "Tokens", name: "opacity/solid" })
-        if (
-          typeof value === "object" &&
-          value !== null &&
-          "collection" in value &&
-          "name" in value
-        ) {
-          const ref = value as { collection: string; name: string };
-          const refVarName = ref.name.replace(/\//g, "-").toLowerCase();
-          return `  --${cssVarName}: var(--${refVarName});`;
-        }
+          // Handle different value types
+          if (typeof value === "string" || typeof value === "number") {
```

Comment:
You have all these naming functions that have some kind of unique naming convention.  Where did you come up with this convention?  Also, the code looks to be basically the same, so I would think it should be a set of common functions to do this, rather than reimplementing every time